backoffice/workflows/api/views.py

```diff
@@ -98,6 +99,56 @@ def resolve(self, request, pk=None):
                 ResolutionDags[serializer.validated_data["value"]].label, pk, extra_data
             )
 
+    @action(detail=True, methods=["post"])
+    def restart(self, request, pk=None):
+
+        params = request.data.get("params", None)
+        restart_current_task = request.data.get("restart_current_task", False)
+
+        workflow = Workflow.objects.get(id=pk)
+
+        data = {"dry_run": False, "dag_run_id": pk, "reset_dag_runs": True}
+
+        executed_dags_for_workflow = {}
+        # find dags that were executed
+        for dag_id in AUTHOR_DAGS[workflow.workflow_type]:
+            response = requests.get(
+                f"{airflow_utils.AIRFLOW_BASE_URL}/api/v1/dags/{dag_id}/dagRuns/{pk}",
+                json=data,
+                headers=airflow_utils.AIRFLOW_HEADERS,
+            )
+            if response.status_code == status.HTTP_200_OK:
+                executed_dags_for_workflow[dag_id] = response.content
+
+        #  assumes current task is one of the failed tasks
+        if restart_current_task:
+
+            data = {"dry_run": False, "dag_run_id": pk, "reset_dag_runs": False, "only_failed": True}
+
+            response = requests.post(
+                f"{airflow_utils.AIRFLOW_BASE_URL}/api/v1/dags/{dag_id}/clearTaskInstances",
+                json=data,
+                headers=airflow_utils.AIRFLOW_HEADERS,
+            )
+            if response.status_code != 200:
+                return Response({"error": "Failed to restart task"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
I would not put in every error `500`, as we have already API exceptions I will create one for failed to restart task for example, with a different error code. `500` is usually for unexpected server error, in these cases we know what's going on.